File: May Challenge 2019/REDONE.py
mod = 1000000007
for _ in range(int(input())):
    n = int(input())

    if n == 1:
        print(n)
    else:
        sum = 1
        for i in range(2, n+1):
            sum = sum + i + sum*(i)
        print(sum % mod)

File: May Challenge 2019/REDONE.cpp
#include<bits/stdc++.h>
using namespace std;

#define lli long long int

int main()
{
    int test; cin >> test;
    lli mod = 1000000007;
    while(test--)
    {
        lli n; cin >> n;
        if(n == 1)
        {
            cout << 1 << endl;
        }
        else
        {
            lli sum = 1;
            for(lli i=2; i<n+1; i++)
                sum = sum + i + sum*i;
            cout << sum % mod << endl;
        }
    }
}